### ISSUE TITLE
exp diff: improvements

### ISSUE THREAD

**Issue by dberenbaum** [iterative/dvc#9147]
We need `dvc exp diff` to provide a report comparing several experiments, and it could be added to a PR.

Some requirements:
- [ ] Accept any number of experiments
- [ ] Include metrics, parameters, and plots
- [ ] Either by flag or default, only include metrics/params values that differ
- [ ] Markdown format with plots embedded as images (may depend on #9121)

Nice to have (out of scope for now):
- [ ] Showing which data files changed
- [ ] Other output formats

Alternatives:
- Studio could generate reports to be added to PRs, which could be more flexible, but it limits where you can use it.

Related: #6451

**Comment by dberenbaum** [comment]
@shcheklein @tapadipti @mattseddon @daavoo The idea here is to have some way to auto-generate a comment to put into PRs that shows useful info about the experiment being merged. What do you think about the proposal above vs possible alternatives?

**Comment by shcheklein** [comment]
sounds good to me, In Studio we can first try to emulate this to better understand what we need. I don't it's that important to have a specific command implemented for Studio, unless I'm missing something.

**Comment by daavoo** [comment]
> The idea here is to have some way to auto-generate a comment to put into PRs that shows useful info about the experiment being merged. What do you think about the proposal above vs possible alternatives?

I think what we are missing is just `dvc plots diff --md` (one of the bullet points there). 
`dvc metrics/params diff` already show only changed by default and support `--md`.

TBH I don't see much value in `exp diff` just being an opinionated alias for merging `dvc metrics/params/plots diff --md`. 

I think it is more flexible to be able to arrange and include titles as you please in the report, we just need to document the pattern.

> Studio could generate reports to be added to PRs, which could be more flexible, but it limits where you can use it.

I do think this alternative could be valuable, even for limited scenarios.

**Comment by daavoo** [comment]
> sounds good to me, In Studio we can first try to emulate this to better understand what we need. I don't it's that important to have a specific command implemented for Studio, unless I'm missing something.

I think what would make sense is for Studio to implement https://github.com/iterative/studio/issues/4987 (and make `Compare` a separate view instead of expandable) so that the URL could be embedded in the P.R.

**Comment by dberenbaum** [comment]
Okay, let's start from the Studio side and see what's needed.

**Comment by dberenbaum** [comment]
Closing since we would need to address in studio and may not need anything in dvc
